extension/js/common/composer.ts

```diff
@@ -548,17 +548,29 @@ export class Composer {
   /**
    * Try to find an intersection of time that public keys of all recipients were usable (if user confirms this with a modal)
    */
-  private encryptMsgAsOfDateIfSomeAreExpiredAndUserConfirmedModal = async (armoredPubkeys: string[]): Promise<Date | undefined> => {
+  private encryptMsgAsOfDateIfSomeAreExpiredAndUserConfirmedModal = async (armoredPubkeys: PubkeyResult[]): Promise<Date | undefined> => {
     const usableUntil: number[] = [];
     const usableFrom: number[] = [];
-    for (const armoredPubkey of armoredPubkeys) {
-      const { keys: [pub] } = await openpgp.key.readArmored(armoredPubkey);
+    const myKeys = armoredPubkeys.filter(ap => ap.isMine);
+    for (const armoredPubkey of armoredPubkeys.filter(k => !myKeys.includes(k))) {
```

Comment:
Actually here please cycle through all keys.

I know you reasoned correctly here, you want to either worry about recipients, or about sender's key, and separate concerns.

But because of the complexity of signatures, their creation time, their expiration time, etc, it's actually important to look at **intersection of all keys**, including our own.

It should not cause trouble with the rest of the logic, because you are testing sender's key first. So if it's actually the sender's key causing trouble, user will always know about it either way.

Maybe what you did will produce the same behavior. I'm not sure. But I'd feel a bit better if it looked at all keys here.

Reply:
That means simply `for (const armoredPubkey of armoredPubkeys) {`

Reply:
Yes, agree, Thanks!